### ISSUE TITLE
Silent Audio Tracks?

### ISSUE THREAD

**Issue by greatmate98** [ace-step/ACE-Step-1.5#286]
I installed the app via uv.  Ironed out all of the kinks (which were for me torchao & triton).  It seems to work fine - i see no errors.  But it just generates two blank audio files, which respect the length i set.  Anyone else encountered this and know a fix?

**Comment by jayvenn21** [comment]
Thanks for the report, silent audio with correct duration usually means the pipeline ran, but audio content was zeroed or never injected.

A few things that would help narrow this down:

1. **Backend / device**
   - Are you running with the default backend (`vllm`) or `--backend pt`?
   - GPU model + CUDA version?

2. **Task type**
   - Does this happen for `text2music` specifically, or also for `cover` / `repaint`?
   - If you switch tasks, does the output remain silent?

3. **LLM enabled**
   - Are you running with the 5Hz LM enabled, or DiT-only mode (LLM disabled)?
   - If you disable the LLM, do you still get silent audio?

4. **Waveform sanity check**
   - If you open one of the generated audio files in an editor (Audacity, etc.), is it truly flat zero amplitude, or just extremely quiet?

From similar reports, common causes tend to be:
- LLM backend silently failing (but DiT still runs → duration correct, content empty)
- Incompatible torch / triton / flash-attn combo causing zeroed outputs
- Running on CPU or fallback paths without explicit warnings

As a quick test, you could try:
```bash
uv run acestep --backend pt

**Comment by DarkAlchy** [comment]
I never get silence (which would be a joy) I get low floor hiss noise when it happens.  The noise we hear under the music that seems to plague this model.

**Comment by greatmate98** [comment]
> Thanks for the report, silent audio with correct duration usually means the pipeline ran, but audio content was zeroed or never injected.
> 
> A few things that would help narrow this down:
> 
> 1. **Backend / device**
>    
>    * Are you running with the default backend (`vllm`) or `--backend pt`?
>    * GPU model + CUDA version?
> 2. **Task type**
>    
>    * Does this happen for `text2music` specifically, or also for `cover` / `repaint`?
>    * If you switch tasks, does the output remain silent?
> 3. **LLM enabled**
>    
>    * Are you running with the 5Hz LM enabled, or DiT-only mode (LLM disabled)?
>    * If you disable the LLM, do you still get silent audio?
> 4. **Waveform sanity check**
>    
>    * If you open one of the generated audio files in an editor (Audacity, etc.), is it truly flat zero amplitude, or just extremely quiet?
> 
> From similar reports, common causes tend to be:
> 
> * LLM backend silently failing (but DiT still runs → duration correct, content empty)
> * Incompatible torch / triton / flash-attn combo causing zeroed outputs
> * Running on CPU or fallback paths without explicit warnings
> 
> As a quick test, you could try:
> 
> uv run acestep --backend pt

Hi thanks for your reply.  To answer your questions:
1)  U gave tried both.  But for whatever reason i found when using the vllm it just generates gibberish.  So have mainly been testing without the vllm
2)  Same results when using cover - i have not tried inpainting
3)  Yes its the same with / without the use of vllm
4)  Yes its a completely flat amplitude. 
5) I tried your suggestion of uv run acestep --backend pt - still no luck.  Other things - im running on a 4080 12GB laptop, 32GB RAM.  Ive tried running on CPU, GPU, switching offload to CPU on/off.  

I originally tried the standalone windows package with zero luck.  Rather than redownload all of the models, after reinstalling via UV i just copied across the models that were downloaded when trying the windows package and placed them in a checkpoints folder, which might be part of the issue?  It seems to detect the models fine.  However i just noticed the following message in my terminal when trying again:

Some weights of the model checkpoint at D:\AI\MusicGen\as15\ACE-Step-1.5\checkpoints\acestep-v15-turbo were not used when initializing AceStepConditionGenerationModel: ['embed_tokens.weight', 'layers.0.input_layernorm.weight', 'layers.0.mlp.down_proj.weight', 'layers.0.mlp.gate_proj.weight'......goes on to list hundreds of weights']
- This IS expected if you are initializing AceStepConditionGenerationModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing AceStepConditionGenerationModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).  

Then this is the log when trying a generation:
2026-02-08 10:10:42.307 | INFO     | acestep.handler:initialize_service:464 - [initialize_service] DiT quantized with: int8_weight_only
2026-02-08 10:12:00.862 | INFO     | acestep.inference:generate_music:387 - [generate_music] LLM usage decision: thinking=True, use_cot_caption=True, use_cot_language=True, use_cot_metas=True, need_lm_for_cot=True, llm_initialized=False, use_lm=False
2026-02-08 10:12:00.862 | INFO     | acestep.handler:generate_music:2846 - [generate_music] Starting generation...
2026-02-08 10:12:00.863 | INFO     | acestep.handler:generate_music:2849 - [generate_music] Preparing inputs...
2026-02-08 10:12:01.271 | INFO     | acestep.handler:_load_model_context:718 - [_load_model_context] Loading vae to cuda
2026-02-08 10:12:01.576 | INFO     | acestep.handler:_load_model_context:731 - [_load_model_context] Loaded vae to cuda in 0.3057s
2026-02-08 10:12:01.692 | INFO     | acestep.handler:_load_model_context:737 - [_load_model_context] Offloading vae to CPU
2026-02-08 10:12:01.971 | INFO     | acestep.handler:_load_model_context:748 - [_load_model_context] Offloaded vae to CPU in 0.2798s
2026-02-08 10:12:02.057 | INFO     | acestep.handler:_prepare_batch:1936 - 
======================================================================
2026-02-08 10:12:02.057 | INFO     | acestep.handler:_prepare_batch:1937 - 🔍 [DEBUG] DiT TEXT ENCODER INPUT (Inference)
2026-02-08 10:12:02.058 | INFO     | acestep.handler:_prepare_batch:1938 - ======================================================================
2026-02-08 10:12:02.059 | INFO     | acestep.handler:_prepare_batch:1939 - text_prompt:
# Instruction
Fill the audio semantic mask based on the given conditions:

# Caption
A peaceful guitar melody

# Metas
- bpm: N/A
- timesignature: N/A
- keyscale: N/A
- duration: 30 seconds
<|endoftext|>

2026-02-08 10:12:02.059 | INFO     | acestep.handler:_prepare_batch:1940 - ======================================================================

2026-02-08 10:12:02.111 | INFO     | acestep.handler:_load_model_context:718 - [_load_model_context] Loading vae to cuda
2026-02-08 10:12:02.397 | INFO     | acestep.handler:_load_model_context:731 - [_load_model_context] Loaded vae to cuda in 0.2845s
2026-02-08 10:12:02.459 | INFO     | acestep.handler:_load_model_context:737 - [_load_model_context] Offloading vae to CPU
2026-02-08 10:12:02.625 | INFO     | acestep.handler:_load_model_context:748 - [_load_model_context] Offloaded vae to CPU in 0.1657s
2026-02-08 10:12:02.626 | INFO     | acestep.handler:preprocess_batch:2147 - [preprocess_batch] Inferring prompt embeddings...
2026-02-08 10:12:02.626 | INFO     | acestep.handler:_load_model_context:718 - [_load_model_context] Loading text_encoder to cuda
`loss_type=None` was set in the config but it is unrecognized. Using the default loss: `ForCausalLMLoss`.
2026-02-08 10:12:03.313 | INFO     | acestep.handler:_load_model_context:731 - [_load_model_context] Loaded text_encoder to cuda in 0.6866s
2026-02-08 10:12:04.216 | INFO     | acestep.handler:preprocess_batch:2150 - [preprocess_batch] Inferring lyric embeddings...
2026-02-08 10:12:04.217 | INFO     | acestep.handler:_load_model_context:737 - [_load_model_context] Offloading text_encoder to CPU
2026-02-08 10:12:05.015 | INFO     | acestep.handler:_load_model_context:748 - [_load_model_context] Offloaded text_encoder to CPU in 0.7980s
2026-02-08 10:12:05.015 | INFO     | acestep.handler:service_generate:2383 - [service_generate] Generating audio...
2026-02-08 10:12:05.016 | INFO     | acestep.handler:_load_model_context:718 - [_load_model_context] Loading model to cuda
2026-02-08 10:12:07.945 | INFO     | acestep.handler:_load_model_context:731 - [_load_model_context] Loaded model to cuda in 2.9288s
2026-02-08 10:12:11.421 | INFO     | acestep.handler:_load_model_context:737 - [_load_model_context] Offloading model to CPU
2026-02-08 10:12:13.974 | INFO     | acestep.handler:_load_model_context:748 - [_load_model_context] Offloaded model to CPU in 2.5516s
2026-02-08 10:12:13.975 | INFO     | acestep.handler:generate_music:2957 - [generate_music] Model generation completed. Decoding latents...
2026-02-08 10:12:14.033 | DEBUG    | acestep.handler:generate_music:2961 - [generate_music] pred_latents: torch.Size([2, 2455, 64]), dtype=torch.bfloat16 pred_latents.min()=tensor(nan, device='cuda:0', dtype=torch.bfloat16), pred_latents.max()=tensor(nan, device='cuda:0', dtype=torch.bfloat16), pred_latents.mean()=tensor(nan, device='cuda:0', dtype=torch.bfloat16) pred_latents.std()=tensor(nan, device='cuda:0', dtype=torch.bfloat16)
2026-02-08 10:12:14.033 | DEBUG    | acestep.handler:generate_music:2962 - [generate_music] time_costs: {'encoder_time_cost': 0.08541131019592285, 'diffusion_time_cost': 2.6330149173736572, 'diffusion_per_step_time_cost': 0.32912686467170715, 'total_time_cost': 2.71842622756958, 'offload_time_cost': 8.000585556030273}
2026-02-08 10:12:14.034 | INFO     | acestep.handler:generate_music:2965 - [generate_music] Decoding latents with VAE...
2026-02-08 10:12:14.034 | INFO     | acestep.handler:_load_model_context:718 - [_load_model_context] Loading vae to cuda
2026-02-08 10:12:14.269 | INFO     | acestep.handler:_load_model_context:731 - [_load_model_context] Loaded vae to cuda in 0.2357s
2026-02-08 10:12:14.272 | DEBUG    | acestep.handler:generate_music:2983 - [generate_music] Before VAE decode: allocated=0.34GB, max=2.62GB
2026-02-08 10:12:14.272 | INFO     | acestep.handler:generate_music:2986 - [generate_music] Using tiled VAE decode to reduce VRAM usage... 
Decoding audio chunks: 100%|█████████████████████████████████████████████████████████████████████████████| 6/6 [00:02<00:00,  2.06steps/s]
2026-02-08 10:12:17.820 | DEBUG    | acestep.handler:generate_music:2993 - [generate_music] After VAE decode: allocated=0.50GB, max=3.08GB 
2026-02-08 10:12:17.906 | INFO     | acestep.handler:_load_model_context:737 - [_load_model_context] Offloading vae to CPU
2026-02-08 10:12:18.102 | INFO     | acestep.handler:_load_model_context:748 - [_load_model_context] Offloaded vae to CPU in 0.1948s
2026-02-08 10:12:18.103 | INFO     | acestep.handler:generate_music:3010 - [generate_music] VAE decode completed. Preparing audio tensors...
2026-02-08 10:12:18.103 | INFO     | acestep.handler:generate_music:3025 - [generate_music] Done! Generated 2 audio tensors.
2026-02-08 10:12:18.818 | DEBUG    | acestep.audio_utils:save_audio:125 - [AudioSaver] Fallback soundfile Saved audio to D:\AI\MusicGen\as15\ACE-Step-1.5\gradio_outputs\batch_1770516738\cea78704-665c-47b5-bf79-ce7395186cd3.mp3 (mp3, 48000Hz)
2026-02-08 10:12:19.589 | DEBUG    | acestep.audio_utils:save_audio:125 - [AudioSaver] Fallback soundfile Saved audio to D:\AI\MusicGen\as15\ACE-Step-1.5\gradio_outputs\batch_1770516738\071dc94c-767b-2179-05c3-c17cea097cde.mp3 (mp3, 48000Hz)
2026-02-08 10:12:19.995 | INFO     | acestep.gradio_ui.events.results_handlers:generate_with_progress:930 - [generate_with_progress] Audio 1 path: D:/AI/MusicGen/as15/ACE-Step-1.5/gradio_outputs/batch_1770516738/cea78704-665c-47b5-bf79-ce7395186cd3.mp3
2026-02-08 10:12:19.995 | INFO     | acestep.gradio_ui.events.results_handlers:generate_with_progress:930 - [generate_with_progress] Audio 2 path: D:/AI/MusicGen/as15/ACE-Step-1.5/gradio_outputs/batch_1770516738/071dc94c-767b-2179-05c3-c17cea097cde.mp3
2026-02-08 10:12:19.997 | INFO     | acestep.gradio_ui.events.results_handlers:generate_with_batch_management:1667 - [generate_with_batch_management] Final yield: 46 core + 9 state


Thanks for your help.

**Comment by jayvenn21** [comment]
Thanks for the detailed logs, this confirms the root cause.

The diffusion step is producing NaN latents (pred_latents.* = nan), which then decode to silent audio. This typically happens with checkpoint/config mismatches (e.g. copying models from the Windows package) or unsupported quantization/backends.

I’ve opened a PR to explicitly detect and error on NaN latents so these failures surface clearly instead of silently producing empty audio.

**Comment by PyrateGFXProductions** [comment]
I am having the same issue after transitioning from Windows to Linux and I am also loading the models from the same remote drive that I was using on the Windows platform. I decided to test if copying the model to the Linux drive would make a difference but it did not. 

Also, this issue does not happen when using the v1_3.5b model located on the same remote drive.

**Comment by DarkAlchy** [comment]
I noticed this from the first moment I installed this and ran it, and I am not sure if this matters, but here it is: DEBUG    | acestep.constrained_logits_processor:_precompute_audio_code_tokens:577 - Found 1535 audio code tokens with values outside valid range [0, 63999]

**Comment by greatmate98** [comment]
> Thanks for the detailed logs, this confirms the root cause.
> 
> The diffusion step is producing NaN latents (pred_latents.* = nan), which then decode to silent audio. This typically happens with checkpoint/config mismatches (e.g. copying models from the Windows package) or unsupported quantization/backends.
> 
> I’ve opened a PR to explicitly detect and error on NaN latents so these failures surface clearly instead of silently producing empty audio.

Thanks mate - i did a fresh download of the models via UV and now everything is working :)
